manifests/init.pp

```diff
@@ -0,0 +1,128 @@
+#
+#   This is set to ``99_simp`` so that regular, alternative, zone manipulation
+#   can occur without interference.
+#
+#   **IMPORTANT:** If this is set to anything besides ``99_simp``, all rules in
+#   this module will **NOT** apply to the default zone! This module is set to
+#   only populate ``99_simp`` zone rules.
+#
+# @param log_denied
+#   What types of logs to process for denied packets.
+#
+#   @see LogDenied in firewalld.conf(5)
+#
+# @param firewall_backend
+#   Allows you to set the backend that firewalld will use.
+#
+#   * Currently set to 'iptables' due to bugs in nftables
+#
+# @param enable_tidy
+#   Enable the ``Tidy`` resources that help keep the system clean from cruft
+#
+# @param tidy_dirs
+#   The directories to target for tidying
+#
+# @param tidy_prefix
+#   The name match to use for tidying files
+#
+# @param tidy_minutes
+#   Number of **minutes** to consider a configuration file 'stale' for the
+#   purposes of tidying.
+#
+# @param simp_zone_interfaces
+#   The network interfaces to which the underlying 99_simp zone should apply
+#
+# @param simp_zone_target
+#   The default target for the 99_simp zone
+#
+class simp_firewalld (
+  Boolean                                              $enable               = 'firewalld' in pick($facts['simplib__firewalls'], 'none'),
+  Boolean                                              $complete_reload      = false,
+  Boolean                                              $lockdown             = true,
+  String[1]                                            $default_zone         = '99_simp',
+  Enum['off', 'all','unicast','broadcast','multicast'] $log_denied           = 'unicast',
+  Enum['iptables','nftables']                          $firewall_backend     = 'iptables',
+  Boolean                                              $enable_tidy          = true,
+  # lint:ignore:2sp_soft_tabs
+  Array[Stdlib::Absolutepath]                          $tidy_dirs            = [
+                                                                                 '/etc/firewalld/icmptypes',
+                                                                                 '/etc/firewalld/ipsets',
+                                                                                 '/etc/firewalld/services'
+                                                                               ],
+  # lint:endignore
+  String[1]                                            $tidy_prefix          = 'simp_',
+  Integer[1]                                           $tidy_minutes         = 10,
+  Array[Optional[String[1]]]                           $simp_zone_interfaces = [],
+  Enum['default', 'ACCEPT', 'REJECT', 'DROP']          $simp_zone_target     = 'DROP'
+) {
+  if $enable {
+    Exec { path => '/usr/bin:/bin' }
+
+    # Upstream module only takes yes/no values
+    $_lockdown_xlat = $lockdown ? { true => 'yes', default => 'no' }
+
+    class { 'firewalld':
+      lockdown         => $_lockdown_xlat,
```

Comment:
Would we want to pass more parameters in here and have them able to be overridden via hiera? One example I can think of initially would be `package_ensure` since with control that across the board with the `simp_options::package_ensure` global flag. It would at least make sense to manage SIMP toggles and other controlled settings as pass-through.

Reply:
Overriding via Hiera works already, you just have to target it at the right class.

I absolutely hate mapping every parameter in the world through classes. It's an extremely fragile tight coupling between components and means that we'll end up having to update this module every time.

There's already a comment at the top that tells you how you what you need to do to override the underlying `firewalld` class.

Reply:
Yeah, I get that as well but in this case it breaks the dynamic of being able to set one simp_options tunable that should affect all SIMP-controlled modules. To move to `latest` at at a minimum now we'd have to have users update `simp_options::package_ensure` and `firewalld::package_ensure`

For other params I'd say that's fine, but this should be coupled if we're treating it as a SIMP component. Otherwise, there's no point from a user status of having the tunable as administrators will not have consistency.

Reply:
Ah...yeah, the package version makes sense, I'll throw that one in there.